File: komilov-mirmakhmud/Core/Proxy/SecurityProxy.php
<?php
require_once __DIR__ . '/../../modules/Security/Camera.php';

class SecurityProxy
{
    private $camera;

    public function __construct()
    {
        $this->camera = new Camera();
    }

    public function accessControl($role)
    {
        if ($role === 'admin') {
            echo $this->camera->monitor();
        } else {
            echo "Access denied. Admins only.\n";
        }
    }
}

File: Raimjonov_Muhammadali/core/proxy/security_proxy.dart
import '../../modules/security/security_module.dart';

class SecurityProxy {
  final SecurityModule _real;
  SecurityProxy(this._real);

  bool requestAccess(String role) {
    if (role == 'admin') {
      return _real.grantAccess();
    } else {
      // guests can only read status
      _real.status();
      return false;
    }
  }
}
